Description: Add libnl-1 support.
 This patch adds libnl support. This patch has been taken from upstream.
Origin: http://dsommers.fedorapeople.org/python-ethtool/python-ethtool-libnl-1.0-support.patch

Index: python-ethtool-0.6/python-ethtool/etherinfo.c
===================================================================
--- python-ethtool-0.6.orig/python-ethtool/etherinfo.c	2011-03-10 10:41:25.282506084 -0500
+++ python-ethtool-0.6/python-ethtool/etherinfo.c	2011-03-10 10:41:21.442500452 -0500
@@ -30,6 +30,52 @@
 #include "etherinfo_struct.h"
 #include "etherinfo.h"
 
+#ifdef LIBNL_1_0
+#define NLHDR_COMMON                            \
+        int                     ce_refcnt;      \
+        struct nl_object_ops *  ce_ops;         \
+        struct nl_cache *       ce_cache;       \
+        struct nl_list_head     ce_list;        \
+        int                     ce_msgtype;     \
+        int                     ce_flags;       \
+        uint32_t                ce_mask;
+
+struct nl_cache
+{
+        struct nl_list_head     c_items;
+        int                     c_nitems;
+        int                     c_iarg1;
+        int                     c_iarg2;
+        struct nl_cache_ops *   c_ops;
+};
+
+struct nl_object
+{
+        NLHDR_COMMON
+};
+
+void nl_cache_free(struct nl_cache *cache)
+{
+        struct nl_object *obj, *tmp;
+
+        if( !cache )
+                return;
+
+        nl_list_for_each_entry_safe(obj, tmp, &cache->c_items, ce_list) {
+                struct nl_cache *cache = obj->ce_cache;
+                if( !cache ) {
+                        break;
+                }
+
+                nl_list_del(&obj->ce_list);
+                obj->ce_cache = NULL;
+                nl_object_put(obj);
+                cache->c_nitems--;
+        }
+        free(cache);
+}
+#endif
+
 /*
  *
  *   Internal functions for working with struct etherinfo
Index: python-ethtool-0.6/setup.py
===================================================================
--- python-ethtool-0.6.orig/setup.py	2011-03-10 10:41:25.322506141 -0500
+++ python-ethtool-0.6/setup.py	2011-03-10 10:41:21.442500452 -0500
@@ -67,7 +67,8 @@
             include_dirs = libnl['include'],
             library_dirs = libnl['libdirs'],
             libraries = libnl['libs'],
-            define_macros = [('VERSION', '"%s"' % version)]
+            define_macros = [('VERSION', '"%s"' % version),
+                             ('LIBNL_1_0', None)]
             )
         ]
 )
